spek-extensions/robolectric/src/main/kotlin/org/spekframework/spek2/robolectric/RobolectricLifeCycleListener.kt

```diff
@@ -0,0 +1,185 @@
+package org.spekframework.spek2.robolectric
+
+import android.R
+import android.os.Build
+import org.robolectric.DefaultTestLifecycle
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.TestLifecycle
+import org.robolectric.android.internal.ParallelUniverse
+import org.robolectric.annotation.Config
+import org.robolectric.internal.ParallelUniverseInterface
+import org.robolectric.internal.SdkEnvironment
+import org.robolectric.internal.bytecode.Sandbox
+import org.robolectric.internal.dependency.CachedDependencyResolver
+import org.robolectric.internal.dependency.DependencyResolver
+import org.robolectric.internal.dependency.LocalDependencyResolver
+import org.robolectric.internal.dependency.PropertiesDependencyResolver
+import org.robolectric.manifest.AndroidManifest
+import org.robolectric.res.Fs
+import org.robolectric.res.PackageResourceTable
+import org.robolectric.res.ResourceMerger
+import org.robolectric.res.ResourcePath
+import org.robolectric.res.ResourceTableFactory
+import org.robolectric.res.RoutingResourceTable
+import org.robolectric.util.Logger
+import org.robolectric.util.ReflectionHelpers
+import org.spekframework.spek2.lifecycle.LifecycleListener
+import org.spekframework.spek2.lifecycle.TestScope
+import java.io.File
+import java.io.IOException
+import java.lang.reflect.InvocationTargetException
+
+class RobolectricLifeCycleListener(
+    val sdkEnvironment: SdkEnvironment,
+    val config:Config,
+    val androidManifest: AndroidManifest): LifecycleListener {
+    private val appResourceTableCache = HashMap<AndroidManifest, PackageResourceTable>()
+    private var compiletimeSdkResourceTable: PackageResourceTable? = null
+    @Transient private var dependencyResolver: DependencyResolver? = null
+
+    var orig: Thread? = null
+    var parallelUniverseInterface: ParallelUniverseInterface? = null
+    var testLifecycle: TestLifecycle<*>? = null
+    var priorContextClassLoader:ClassLoader? = null
+
+    override fun beforeExecuteTest(test: TestScope) {
+        super.beforeExecuteTest(test)
+        priorContextClassLoader = Thread.currentThread().contextClassLoader
+        Thread.currentThread().contextClassLoader = sdkEnvironment.robolectricClassLoader
+
+        parallelUniverseInterface = getHooksInterface(sdkEnvironment)
+        val cl = sdkEnvironment.bootstrappedClass<TestLifecycle<*>>(getTestLifecycleClass())
+        testLifecycle = ReflectionHelpers.newInstance<TestLifecycle<*>>(cl)
+
+        parallelUniverseInterface!!.setSdkConfig(sdkEnvironment.sdkConfig)
```

Comment:
Just enforce it right after initialization :)

Reply:
fixed.